presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroRowDecoder.java

```diff
@@ -31,11 +34,16 @@
 import static com.google.common.collect.ImmutableMap.toImmutableMap;
 import static io.prestosql.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
 import static java.util.Objects.requireNonNull;
+import static org.apache.avro.file.DataFileConstants.MAGIC;
 
 public class AvroRowDecoder
         implements RowDecoder
 {
     public static final String NAME = "avro";
+    public static final int HEADER_LENGTH = 1 + Integer.BYTES;
+
+    private static ThreadLocal<BinaryDecoder> reuseDecoder = ThreadLocal.withInitial(() -> null);
```

Comment:
Why do we need thread local? Is `BinaryDecoder` thread safe?

Reply:
BinaryDecoder is not threadsafe. There is a configure() method which sets the data.